README.md
GoldFash Encrypto

=============


* Tags: Racean Ford, GoldFash Hosting, encrypto, encryption, secure, java, administration, panel, console

* Requires at least: 3.9.2

* Tested up to: 3.9.2

* Stable Tag: 1.0

Easily Encrpyt Content in your wordpress pages or post by using a simple shortcode "[gold_secure] Content Here [/gold_secure]" 

== Description ==

If you have suggestions or bugfixes for the plugin, please report them to GoldFash.

**Languages**

This plugin has been translated into the following languages:

1. English

If you want to translate the plugin into your language, [contact me](http://goldfash.com/contact).

== Installation ==

1. Activate the plugin
2. Contact GoldFash Hosting (302-387-4653 EXT 1) request a license key & security secret code.
3. Input your license key in license.php from "plugin editor". 
4. Use short code [gold_secure] The greatness of all comes from within.[/gold_secure]
2. Take a coffee and relax, you did a great job!


== Screenshots ==

1. <img src="


== Changelog ==

= 0.1 =

* First release!

== Upgrade Notice ==

= 1.3 =
Updated a few security bugs


CHANGES.md
== Changelog ==

= 0.1 =

* First release!



== Current Upgrade Notice ==

= 0.1 =

== #TODO ==
* Cache
* Additional Settings
* Additional help options
* Add-on Plugins Area


goldsecure.php
<?php
/*
Plugin Name: GoldFash Secure Encryption
Plugin URL: https://goldfash.com:443/plugins
Description: Gold Secure Encryption encrypts any text (if you want) on server and decrypts it on client (using javascript) to avoid your email and any other sensitive content being understood by robots and net filters. Simply add [gold_secure]...[/gold_secure] shortcode to encrypt your blog.
Version: 1.0.3.1
Author: GoldFash Design
Author URI:        https://goldfash.com:443/
Contributors:      raceanf
Domain Path:       /languages
Text Domain:       Gold-Secure-Encryption
GitHub Plugin URI: https://github.com/goldfashhosting/Gold-Secure-Encryption
GitHub Branch:     master
*/
require('functions.php');
$gold_secure_path = NULL;
$gold_secure_script_handle = 'gold_secure_script';
$gold_secure_js_name = 'gold_secure.js';

add_action('init', 'gold_secure_init');

add_shortcode('gold_secure', 'gold_secure_process');

function gold_secure_init(){
    global $gold_secure_path;
    global $gold_secure_js_name;
    global $gold_secure_script_handle;
    
    $gold_secure_path = WP_PLUGIN_URL . '/' . str_replace(basename(__FILE__), "", plugin_basename(__FILE__));
    wp_enqueue_script($gold_secure_script_handle);
    
}

function gold_secure_ord($str, $len = -1, $idx = 0, &$bytes = 0){
    if ($len == -1){
        $len = strlen(str);
    }
    $h = ord($str{$idx});

    if ($h <= 0x7F) {
        $bytes = 1;
        return $h;
    }
    else if ($h < 0xC2) {
        return false;
    }
    else if ($h <= 0xDF && $idx < $len - 1) {
        $bytes = 2;
        return ($h & 0x1F) <<  6 | (ord($str{$idx + 1}) & 0x3F);
    }
    else if ($h <= 0xEF && $idx < $len - 2) {
        $bytes = 3;
        return ($h & 0x0F) << 12 | (ord($str{$idx + 1}) & 0x3F) << 6
                                 | (ord($str{$idx + 2}) & 0x3F);
    }           
    else if ($h <= 0xF4 && $idx < $len - 3) {
        $bytes = 4;
        return ($h & 0x0F) << 18 | (ord($str{$idx + 1}) & 0x3F) << 12
                                 | (ord($str{$idx + 2}) & 0x3F) << 6
                                 | (ord($str{$idx + 3}) & 0x3F);
    }    
    return false;
}

function gold_secure_unicode($dec) {
	$hex = dechex($dec);
	if ($dec < 16) {
		return '\\x0' . $hex;
	}
	if ($dec < 256) {
		return '\\x' . $hex;
	}
	if ($dec < 4096) {
		return '\\u0' . $hex;
	}
	
	return '\\u' . $hex;
}
require('license.php');
function gold_secure_encode($content, $text = ""){
    if ($content == NULL || is_feed()){
        return $text;
    }
    
    $len = strlen($content);
    
    if ($len == 0){
        return "";
    }
    
    $idx = 0;
    
    $ord = gold_secure_ord($content, $len, $idx, $idx);
    $script = gold_secure_unicode($ord);
    while ( $idx < $len){
        $bytes = 0;
        $script .= gold_secure_unicode(gold_secure_ord($content, $len, $idx, $bytes));
        $idx += $bytes;
    }
    
    $divid = "https://goldFash.com/Encrypto.seci-APPDiv";
    
    for ($i = 0; $i < 5; ++$i) {
        $divid .= rand(2014, 173905);
    }

    $js = "<!-- Secure Encryption Provided by www.GoldFash.com --!><span id='$divid.aspx' APPid='GoldFash_Secure-Ecrpyto'>$text</span><script type='text/javascript'>var x = document.getElementById('$divid.aspx');x.parentNode.removeChild(x);document.write('$script');</script> <!-- Secure Encryption Provided by www.GoldFash.com --!><!-- #Credits -->
<!-- // GoldFash.com Hosting \\
// Website Encryption Owned by RaFco, A Family Co \\
// GoldFash Hosting and www.rafco.us
// © RaFco|AFC - RFIG|AFC - GoldFash \\
// CEO - Racean Ford | facebook.com/racean.ford | facebook.com/rafcoafc \\ --!>
<!-- www.rafco.us is hosted by GoldFash Hosting visit us online www.goldfash.com --!>
	";
    
    return $js;
}

function gold_secure_process($attr, $content = NULL){
    extract(
        shortcode_atts(
            array(
                'text' => ''
            ),
            $attr
        )
    );
    return gold_secure_encode($content, $text);
}

?>


ee_con/GoldFash/EGC-L.php
<?php
require('functions.php');
$ecgtr = ($goldECSecurity);
$eceLiString = convert_uuencode($ecgtr);
$ecDelIrG = convert_uudecode($eceLiString);
$ecinas = gethostbyname($_SERVER['SERVER_NAME']);
$ecgtra = ($ecinas);
$eceLiStringa = convert_uuencode($ecgtra);
$ecDelIrGa = convert_uudecode($eceLiStringa);
$ecserver_hostname = ($ecDelIrGa); 
$api_endpoint="{$ecargs}{$ecmecaG}{$ecbgaz}{$eczabe}{$ecxzta}{$ecDelIrG}{$ecssarp}{$ecgtri}{$ecsssr}{$ecserver_hostname}";
$key_data=file_get_contents($api_endpoint);
$url_decoded_key_data=urldecode($key_data);
$base64_decoded_key_data=base64_decode($url_decoded_key_data);
list($response, $valid_checksum) = explode('{spbas}', $ebase64_decoded_key_data);
if (md5($hashECAuth . $response)!=$valid_checksum) {
    die('Please Contact <a href="https://www.goldfash.com/G3TGOLD.php?consENTRY!=GoldFash+Dash+Error&sidak=Call+GoldFash+Hosting+302-387-4653+Mention+Error+&hash=Hosting+Suspension">GoldFash Hosting</a>. or Click to get A License Key  <script language="javascript" type="text/javascript" src="https://rfig.us/A-GOLD/plugin.stream/cx.VERF./GeT/e.X.5/order/buy_now.php?product=72d931b3"></script>
 [CODE B100C]');
}
$key_data=unserialize($response);
echo '<pre>';
echo '</pre>';
